zebra-chain/src/transparent.rs

```diff
@@ -132,6 +134,28 @@ impl Input {
             unreachable!("unexpected variant: Coinbase Inputs do not have OutPoints");
         }
     }
+
+    /// Get the value balance of this input.
+    pub fn value_balance(
+        &self,
+        ordered_utxos: &HashMap<OutPoint, utxo::OrderedUtxo>,
+    ) -> Amount<NegativeAllowed> {
+        match self {
+            Input::PrevOut { outpoint, .. } => {
+                let utxos = utxos_from_ordered_utxos(ordered_utxos.clone());
+                if utxos.contains_key(outpoint) {
+                    utxos[outpoint]
+                        .output
+                        .value
+                        .constrain()
+                        .expect("conversion from NonNegative to NegativeAllowed is always valid")
+                } else {
+                    panic!("Provided Utxos don't have transaction Outpoint")
+                }
```

Comment:
Optional: I think this simplifies down to something like:
```suggestion
                ordered_utxos.get(outpoint)
                    .expect("Provided Utxos don't have transaction Outpoint")
                    .utxo
                    .output
                    .value
                    .constrain()
                    .expect("conversion from NonNegative to NegativeAllowed is always valid")
```